### PR TITLE
grub2: Update to 2.06-rc1

### PR DESCRIPTION
**Pull request by christopherco**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

Update grub2 from 2.02 to 2.06-rc1 which handles BootHole v2. Additionally, we
drop all previous patches and rebaseline using a minimal number of patches
from FC34. These patches implement Secure Boot Handover protocol (needed
so the TPM Eventlog can be exposed to the kernel for TPM attestation scenarios)
and a few other nice-to-have fixes.

2.06 also introduces a new generation number based revocation mechanism known
as Secure Boot Advanced Targeting (SBAT) into the grub EFI binary. Components
that utilize the SHIM for secure boot will add an .sbat field into their binary's
PE-header, allowing the SHIM to check the component's sbat field against known
good component versions and allow for version-based revocation.

###### Does this affect the toolchain?  

NO

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2020-14372
- https://nvd.nist.gov/vuln/detail/CVE-2020-25632
- https://nvd.nist.gov/vuln/detail/CVE-2020-25647
- https://nvd.nist.gov/vuln/detail/CVE-2020-27749
- https://nvd.nist.gov/vuln/detail/CVE-2020-27779
- https://nvd.nist.gov/vuln/detail/CVE-2021-3418
- https://nvd.nist.gov/vuln/detail/CVE-2021-20225
- https://nvd.nist.gov/vuln/detail/CVE-2021-20233  

###### Test Methodology

Local build, pipeline build. Local testing

### CODE REVIEW

File: SPECS-SIGNED/grub2-efi-binary-signed-aarch64/grub2-efi-binary-signed-aarch64.spec [hunk 1]
```diff
@@ -1,8 +1,8 @@
 %global debug_package %{nil}
 Summary:        Signed GRand Unified Bootloader for aarch64 systems
 Name:           grub2-efi-binary-signed-aarch64
-Version:        2.02
-Release:        26%{?dist}
+Version:        2.06
```
Comment (PawelWMS) [inline SPECS-SIGNED/grub2-efi-binary-signed-aarch64/grub2-efi-binary-signed-aarch64.spec:4]: Shouldn't this be `2.06-rc1`?

Comment (christopherco) [reply SPECS-SIGNED/grub2-efi-binary-signed-aarch64/grub2-efi-binary-signed-aarch64.spec:4]: Don't believe I can place `-` in the Version number field. I debated to put `2.06~rc1` but wasn't sure what would happen when 2.06 was officially released and our tools did version comparison checks.

Comment (christopherco) [reply SPECS-SIGNED/grub2-efi-binary-signed-aarch64/grub2-efi-binary-signed-aarch64.spec:4]: I'll see if I can construct a small test for this. I would prefer to use `2.06~rc1` here if I can.

Comment (PawelWMS) [reply SPECS-SIGNED/grub2-efi-binary-signed-aarch64/grub2-efi-binary-signed-aarch64.spec:4]: Right, the tilde makes a lot more sense, sorry.

Comment (christopherco) [reply SPECS-SIGNED/grub2-efi-binary-signed-aarch64/grub2-efi-binary-signed-aarch64.spec:4]: Okay testing shows `2.06` > `2.06~rc1` > `2.02`, which is the intended behavior. Will update this PR with `2.06~rc1`

Comment (christopherco) [reply SPECS-SIGNED/grub2-efi-binary-signed-aarch64/grub2-efi-binary-signed-aarch64.spec:4]: https://docs.fedoraproject.org/en-US/packaging-guidelines/Versioning/#_versioning_prereleases_with_tilde

Comment (PawelWMS) [reply SPECS-SIGNED/grub2-efi-binary-signed-aarch64/grub2-efi-binary-signed-aarch64.spec:4]: Thanks!

**Review by PawelWMS — APPROVED** [review]

**Review by rlmenge — APPROVED** [review]

**Pull request merged**